### PR TITLE
Implement backend support for vendor cards

### PR DESCRIPTION
**Pull request by arsalansufi**
## Overview

Issue link: https://github.com/votingworks/vxsuite/issues/4760

This PR implements backend support for vendor cards. Vendor cards will be used by VotingWorks employees to access low-level vendor functionality, currently spread across the TTY2 menu and sys admin UI (e.g., rebooting to the BIOS, recreating a machine's cert, etc.). Once we've fully built things out, vendor cards should render the TTY2 menu obsolete and should also allow us to remove functionality that we've shoehorned into the sys admin UI but that doesn't actually belong there. This will also address the VVSG2 certification issue posed by the TTY2 menu's lack of two-factor auth.

Also included in this PR are scripts to program vendor cards, documented in the libs/auth README. Unlike the other card types, vendor cards cannot be programmed by a VxAdmin. They can only be programmed by VotingWorks directly. They can be jurisdiction-specific or jurisdiction-agnostic and expire after 7 days.

This PR doesn't yet implement the vendor card UI. This will likely consist of a single button to boot into the vendor menu, which will look like the existing TTY2 menu. That'll come in subsequent PRs.

## Testing Plan

- [x] Updated libs/auth tests, maintaining 100% code coverage
- [x] Manually tested all existing and new libs/auth scripts
- [x] Manually tested the updated dev dock

## Checklist

- [x] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced

### CODE REVIEW

File: libs/auth/scripts/configure-dev-java-card
Comment (arsalansufi) [inline libs/auth/scripts/configure-dev-java-card:7]: Made some updates to how the various libs/auth scripts use env vars, making the scripts more consistent with each other.

No updates to vxsuite-complete-system necessary. And all scripts can still be called as they were, the only exception being the exact command for initial Java Card configuration for production cards. I document that updated command in a subsequent README-update commit. I'll also update other relevant documentation after I merge this PR and share an update in #vxsuite-prodeng

File: libs/auth/scripts/src/configure_java_card.ts
Comment (arsalansufi) [inline libs/auth/scripts/src/configure_java_card.ts:289]: Moved env var reading from outside `main` into `main` in all relevant scripts. One positive side effect of this is cleaner error messages for missing env vars.

_Before_

```
$ ./scripts/configure-java-card 

/home/vx/code/vxsuite/libs/basics/build/assert.js:10
        throw new Error(message);
              ^
Error: Missing required NODE_ENV env var
    at assert (/home/vx/code/vxsuite/libs/basics/build/assert.js:10:15)
    at getRequiredEnvVar (/home/vx/code/vxsuite/libs/auth/src/env_vars.ts:10:3)
    at getNodeEnv (/home/vx/code/vxsuite/libs/auth/src/env_vars.ts:19:19)
    at isNodeEnvProduction (/home/vx/code/vxsuite/libs/auth/src/env_vars.ts:31:10)
    at shouldUseProdCerts (/home/vx/code/vxsuite/libs/auth/src/config.ts:25:10)
    at constructJavaCardConfigForVxProgramming (/home/vx/code/vxsuite/libs/auth/src/config.ts:119:28)
    at Object.<anonymous> (/home/vx/code/vxsuite/libs/auth/scripts/src/configure_java_card.ts:36:11)
    at Module._compile (node:internal/modules/cjs/loader:1191:14)
    at Module._compile (/home/vx/code/vxsuite/node_modules/.pnpm/source-map-support@0.5.21/node_modules/source-map-support/source-map-support.js:568:25)
    at Module.mod._compile (/home/vx/code/vxsuite/node_modules/.pnpm/esbuild-runner@2.2.2_esbuild@0.17.11/node_modules/esbuild-runner/src/hook.ts:53:22)
```

_After_

```
$ ./scripts/configure-java-card 
❌ Missing required NODE_ENV env var
```

File: libs/auth/scripts/src/utils.ts
Comment (arsalansufi) [inline libs/auth/scripts/src/utils.ts:48]: Prepping this helper for use in the vendor card creation script, introduced in the next commit

File: libs/dev-dock/frontend/src/dev_dock.tsx
Comment (arsalansufi) [inline libs/dev-dock/frontend/src/dev_dock.tsx:216]: You can see here what the dev dock looks like with one more card option:

dev-dock

File: apps/admin/frontend/src/components/app_routes.tsx
Comment (arsalansufi) [inline apps/admin/frontend/src/components/app_routes.tsx:95]: A placeholder. Will be adding the actual UI in my next PR

File: libs/auth/src/java_card.ts
Comment (arsalansufi) [reply libs/auth/src/java_card.ts:109]: This is already documented in existing code comments, just synthesizing

**Comment by arsalansufi** [comment]
@benadida The most relevant commits for security review are https://github.com/votingworks/vxsuite/pull/4827/commits/172742f658a7f90d5e83b2dc68dd95c04dd0e829 + https://github.com/votingworks/vxsuite/pull/4827/commits/38a0d4866bb07fa775576aadc3bebce410b325ea + https://github.com/votingworks/vxsuite/pull/4827/commits/4bd3508b6ef6eeb960725247bc5bc2fc8de1d40c

File: libs/auth/src/java_card.ts
Comment (benadida) [inline libs/auth/src/java_card.ts:473]: this was the main thing I was looking for :)

**Review by benadida — COMMENTED** [review]
approved from security review perspective.

File: libs/auth/README.md
Comment (jonahkagan) [inline libs/auth/README.md:1]: There's still at least one reference to the Bakery in here. Might want to find-and-replace with "VotingWorks facility" like you did elsewhere

File: apps/admin/frontend/src/components/smartcard_modal/smartcard_modal.test.tsx
Comment (jonahkagan) [inline apps/admin/frontend/src/components/smartcard_modal/smartcard_modal.test.tsx:188]: why is the election definition prompt displayed for vendor cards?

**Review by jonahkagan — APPROVED** [review]
LGTM!

File: apps/admin/frontend/src/components/smartcard_modal/smartcard_modal.test.tsx
Comment (arsalansufi) [reply apps/admin/frontend/src/components/smartcard_modal/smartcard_modal.test.tsx]: Ah yeah, no good reason. I initially thought that it was fine for that message to get displayed when there's no election definition, even if the card you just inserted is a vendor card. But on second thought, that could be confusing when you can't actually program vendor cards on a VxAdmin. Updated accordingly: https://github.com/votingworks/vxsuite/pull/4827/commits/625d03859143d168c73d79d01ba213c08a4446ed

**Pull request merged**
